### DISCUSSION TITLE
Model `find` returns result with partial matching

### DISCUSSION THREAD

**Discussion by pokhiii** [laravel/framework#39245] [Q&A]
- Laravel Version: 8.61.0
- PHP Version: 7.4.22
- Database Driver & Version: mysql 5.7.34

### Description:
Model `find` returns a result with partial matching. See the screenshot below. Should the expected result be `null`?

### Steps To Reproduce:

![image](https://user-images.githubusercontent.com/11808845/137707364-5780ff35-8660-4d06-8e6e-49b768a4ffe5.png)

**Comment by robert-moore96** [comment]
looks like somewhere is casting the input to an integer

**Reply by pokhiii** [reply]
Yeah, found out it was casting

```php
(int) "22nwjfdja" = 22
```
